### ISSUE TITLE
Cannot update optional fields to Null to remove them.

### ISSUE THREAD

**Issue by lawmicha** [aws-amplify/amplify-category-api#2562]
Amplify CLI version 12.10.3

Schema
```
type User2 @model @auth(rules: [{allow: public, operations: [read]}, {allow: private, operations: [read]}, {allow: owner, operations: [create, update]}]) {
    id: ID!
    requiredName: String!
    optionalName: String
} 
```

Semd `createUser2` request, with the user pool user signed in ("michael" user)
```
mutation CreateUser {
  createUser2(input: {requiredName: "1", optionalName: "2"}) {
    createdAt
    optionalName
    id
    owner
    requiredName
    updatedAt
  }
}
```

Response
```
{
  "data": {
    "createUser2": {
      "createdAt": "2024-05-16T16:01:13.753Z",
      "optionalName": "2",
      "id": "a74354c6-afa4-4f8f-8861-79b2f25cd9d7",
      "owner": "michael",
      "requiredName": "1",
      "updatedAt": "2024-05-16T16:01:13.753Z"
    }
  }
}
```

Let's say now I want to remove the value set for `optionalName`. Perform an update on the user with `optionalName` set to null.
```
mutation UpdateOptionalFieldToNull {
  updateUser2(input: {id: "a74354c6-afa4-4f8f-8861-79b2f25cd9d7", optionalName: null}) {
    createdAt
    id
    optionalName
    owner
    requiredName
    updatedAt
  }
}
```

Response
```
{
  "data": {
    "updateUser2": null
  },
  "errors": [
    {
      "path": [
        "updateUser2"
      ],
      "data": null,
      "errorType": "Unauthorized",
      "errorInfo": null,
      "locations": [
        {
          "line": 29,
          "column": 3,
          "sourceName": null
        }
      ],
      "message": "Unauthorized on [optionalName]"
    }
  ]
}
```

The issue is that I cannot update the user to remove an optional string value, whether it was originally set to something or not.

_Originally posted by @lawmicha in https://github.com/aws-amplify/amplify-swift/issues/3677#issuecomment-2115633689_

**Comment by lawmicha** [comment]
is this because  `operations: [create, update]}` does not include `delete` so the customer cannot remove fields that were set? or is the delete operation only applies to `deleteUser` operation?

**Comment by palpatim** [comment]
tl;dr: You need to give the owner delete permissions on the field while continuing to restrict delete on the model.

See https://github.com/aws-amplify/amplify-category-api/issues/2487#issuecomment-2103500393 for longer discussion.

**Comment by lawmicha** [comment]
Thanks @palpatim, this was a customer from https://github.com/aws-amplify/amplify-swift/issues/3677 @dan-codes1 let us know if you have anymore questions

**Comment by dan-codes1** [comment]
I understand the issue now. Thanks @lawmicha

**Comment by AnilMaktala** [comment]
Thank you for confirming @dan-codes1. We are closing the issue.
